registry/quilt_server/views.py

```diff
@@ -2307,6 +2316,171 @@ def _comment_dict(comment):
         contents=comment.contents
     )
 
+@app.route('/api/users/set_role', methods=['POST'])
+@api(enabled=ENABLE_USER_ENDPOINTS, require_admin=True, schema=USERNAME_ROLE_SCHEMA)
+@as_json
+def set_role():
+    """
+    Manages the role attached to a user.
+
+    The request should contain a JSON object with two keys:
+        username(string): username of user to edit
+        role(string): name of role to attach to user
+
+    A user can only have one role at a time.
+    To remove a role from a user, set role_name to the empty string.
+    """
+    data = request.get_json()
+    username = data['username']
+    role_name = data['role']
+    user = User.query.filter_by(name=username).with_for_update().one_or_none()
+    if user is None:
+        raise ApiException(requests.codes.bad_request,
+                           "No user exists by the provided name.")
+    if role_name == '':
+        # delete role from user
+        user.role_id = None
+    else:
+        role = Role.query.filter_by(name=role_name).one_or_none()
+        if role is None:
+            raise ApiException(requests.codes.bad_request,
+                               "No role exists by the provided name.")
+        user.role_id = role.id
+
+    db.session.add(user)
+    db.session.commit()
+    return {}
+
+def _role_dict(role):
+    role_dict = {
+        'id': role.id,
+        'name': role.name,
+        'arn': role.arn
+    }
+    return role_dict
+
+@app.route('/api/roles', methods=['POST'])
```

Comment:
Should have a trailing slash - since "roles" is a group, and we're creating a new item

Reply:
I don't think that is correct for rest semantics:
https://hackernoon.com/restful-api-designing-guidelines-the-best-practices-60e1d954e7c9
https://docs.microsoft.com/en-us/azure/architecture/best-practices/api-design